client/zweb/deployer/index.js

```diff
@@ -146,13 +146,35 @@ class Deployer {
         console.log('Deploy finished');
     }
 
-    async deployContract(target, contractName, fileName) {
+    static async getPragmaVersion(fileName){
```

Comment:
@Tunji17 Just a suggestion... since the `deployContract` function reads the entire contents of the smart contract, perhaps its possible to simply pass the source directly to the `getPragmaVersion` function and then perform a regx on that string, something like this:

```
    static async getPragmaVersion(source){
        let regex = /pragma solidity [\^\~\>\<]?=?(?<version>[0-9\.]*);/;
        let found = null
        if (found = source.match(regex)) {
            return found.groups.version
        } else {
            throw new Error('Contract has no compiler version')
        }
    }
```

Reply:
i'll test this out and get back to you on this

Reply:
OK thank you. What you have done works of course. In order to implement the function like this you would need to read the source of the contract in the `deployContract` before calling `getPragmaVersion` and store the source in a variable to then use in the `compileConfig` object. It should be pretty straightforward to change, but if it causes issues. then we can leave it.

Reply:
Yeah I understand